File: Java/maze/EnchantedMazeFactory.java
package maze;

public class EnchantedMazeFactory extends MazeFactory {
	public EnchantedMazeFactory(){
		super();
	}
	public Room makeRoom(int n){
		return new EnchantedRoom(n, castSpell());
	}
	public Door makeDoor(Room r1, Room r2){
		return new DoorNeedingSpell(r1,r2);
	}
	protected Spell castSpell(){
		return new Spell();
	}
}

File: Python/EnchantedMazeFactory.py
from MapSite import EnchantedRoom, Wall, DoorNeedingSpell
from Maze import Maze
from MazeFactory import makeMaze, makeWall
from Spell import Spell

def makeRoom(n):
    def castSpell():
        return Spell()
    return EnchantedRoom(n, castSpell())

def makeDoor(r1, r2):
    return DoorNeedingSpell(r1, r2)
